android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -82,14 +82,35 @@ protected void onPreExecute() {
         diff = numRepeatingGroups - currNumRepeatingGroups;
         for (int i = 0; i < diff; i++) {
             try {
+
                 repeatingGroups.add(buildRepeatingGroupLayout(parent));
+
             } catch (Exception e) {
                 Timber.e(e);
             }
         }
+
+        updateRepeatingGrpCountObject();
+
         return repeatingGroups;
     }
 
+    private void updateRepeatingGrpCountObject() {
+        try {
+            JSONObject countFieldObject = Utils.getRepeatingGroupCountObj(widgetArgs);
+            if (countFieldObject != null) {
+                int currentCount = numRepeatingGroups;
+                String strNumOfRepeatedGroups = countFieldObject.optString(VALUE);
+                if (StringUtils.isNotBlank(strNumOfRepeatedGroups)) {
+                    currentCount += Integer.parseInt(strNumOfRepeatedGroups);
```

Comment:
Not sure why this is `+=`. In the case that the value in the reference edit text changes, shouldn't the new value be the new number of repeating groups generated? 

The logic atm looks at the difference and either generates extra groups for a +ve diff or deletes excess groups for a -ve diff.

Reply:
Ideally the new value should be the summation of all the groups generated otherwise the count object will have incorrect value.